File: structural/flyweight/python/concrete_flyweight.py
from interface import Shape


class Circle(Shape):
    def __init__(self, color: str) -> None:
        self.color = color

    def draw(self, x: int, y: int, radius: int) -> None:
        print(f"Circle: Draw() [Color : {self.color}, x : {x}, y : {y}, radius : {radius}]")

File: structural/flyweight/typescript/concrete-flyweight.ts
import { Shape } from "./interface";

export class Circle implements Shape {
    private color: string;

    constructor(color: string) {
        this.color = color;
    }

    draw(x: number, y: number, radius: number): void {
        console.log(`Circle: Draw() [Color : ${this.color}, x : ${x}, y : ${y}, radius : ${radius}]`);
    }
}